snuba/subscriptions/scheduler_processing_strategy.py

```diff
@@ -251,6 +255,17 @@ def submit(self, message: Message[Tick]) -> None:
             self.__next_step.submit(self.__buffers[tick_partition].popleft())
             return
 
+        # Periodically record the legnth of each buffer
+        now = time.time()
+        if now - self.__last_recorded_time > self.__record_frequency_seconds:
+            for partition_index in self.__buffers:
+                self.__metrics.gauge(
+                    "TickBuffer.queue_size",
+                    len(self.__buffers[partition_index]),
+                    tags={"partition": str(partition_index)},
+                )
+            self.__last_recorded_time = now
+
```

Comment:
This should be in its own function

Reply:
I'll leave it up to you of course but as an external reader I don't find this more readable. Most importantly, being able to separate the observability logic from the application logic is important. Looking at the function, it's not obvious at a glance what is doing what. 

This function also does not fit into my monitor which makes its purpose and logic harder to follow.

Reply:
Ok, I will move it out